### ISSUE TITLE
CS0231 error, when called, also yields a misleading CS1503 error

### ISSUE THREAD

**Issue by chucker** [dotnet/roslyn#18875]
**Version Used**:  15.1 26403.7 Release

**Steps to Reproduce**:

1. Write a method `Foo` with a params parameter that isn't the last parameter.
2. Call this method from a second method `Bar`.

For instance:

```csharp
        public static void Foo(params int[] vals, bool truth)
        {

        }

        public static void Bar()
        {
            Foo(1, true);
        }
```

**Expected Behavior**:

This should yield a single error, in `Foo`:

> Error	CS0231	A params parameter must be the last parameter in a formal parameter list

**Actual Behavior**:

It yields an additional, misleading error in `Bar`:

> Error	CS1503	Argument 1: cannot convert from 'int' to 'int'

**Comment by davkean** [comment]
@grafter This looks like a dupe of https://github.com/dotnet/roslyn/issues/12933?

**Comment by AlekseyTs** [comment]
Should be fixed in https://github.com/dotnet/roslyn/pull/18923
